example/gmres/test_cmplx_A.cpp

```diff
@@ -0,0 +1,147 @@
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Jennifer Loe (jloe@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include<math.h>
+#include"KokkosKernels_IOUtils.hpp"
+#include<Kokkos_Core.hpp>
+#include<Kokkos_Random.hpp>
+#include<KokkosBlas.hpp>
+#include<KokkosBlas3_trsm.hpp>
+#include<KokkosSparse_spmv.hpp>
+
+#include"gmres.hpp"
+
+int main(int /*argc*/, char ** /*argv[]*/) {
+
+  typedef Kokkos::complex<double>           ST;
+  typedef int                               OT;
+  typedef Kokkos::DefaultExecutionSpace     EXSP;
+
+  using ViewVectorType = Kokkos::View<ST*,Kokkos::LayoutLeft, EXSP>;
+
+  std::string filename("young1c.mtx"); // example matrix
+  std::string ortho("CGS2"); //orthog type
+  int m = 100; //Max subspace size before restarting.
+  double convTol = 1e-05; //Relative residual convergence tolerance.
+  int cycLim = 60;
+
+  std::cout << "File to process is: " << filename << std::endl;
+  std::cout << "Convergence tolerance is: " << convTol << std::endl;
+
+  //Initialize Kokkos AFTER parsing parameters:
+  Kokkos::initialize();
+  {
+
+  // Read in a matrix Market file and use it to test the Kokkos Operator.
+  KokkosSparse::CrsMatrix<ST, OT, EXSP> A = 
+    KokkosKernels::Impl::read_kokkos_crst_matrix<KokkosSparse::CrsMatrix<ST, OT, EXSP>>(filename.c_str()); 
+
+  int n = A.numRows();
+  ViewVectorType X("X",n); //Solution and initial guess
+  ViewVectorType Wj("Wj",n); //For checking residuals at end.
+  ViewVectorType B(Kokkos::ViewAllocateWithoutInitializing("B"),n);//right-hand side vec
```

Comment:
Same as above comment on deprecation of `Kokkos::ViewAllocateWithoutInitializing`